src/main/java/org/terasology/Lakes/LakeProvider.java

```diff
@@ -44,16 +50,24 @@ public void process(GeneratingRegion region) {
 
         Border3D border = region.getBorderForFacet(LakeFacet.class);
         //Extend border by max radius + max length + max outer length and max lakedepth
-        border = border.extendBy(11, 12, 15);
+        border = border.extendBy(Lake.MAX_DEPTH, Lake.MAX_DEPTH, (Lake.MAX_SIZE));
         LakeFacet lakes = new LakeFacet(region.getRegion(), border);
         Region3i worldRegion = lakes.getWorldRegion();
+        BiomeFacet biomeFacet = region.getRegionFacet(BiomeFacet.class);
+
         for (Vector3i pos : worldRegion) {
             float sHeight = surfaceHeightFacet.getWorld(pos.x(), pos.z());
-            float noiseValue = noise.noise(pos.x(), pos.y(), pos.z());
-            if (pos.y() < sHeight - 20 && noiseValue > 0.99998) {
-                lakes.add(new Lake(pos, 10 + 20 * Math.abs(Math.round(noise.noise(pos.x(), pos.z())))));
-            } else if (pos.y() == Math.round(sHeight) && noiseValue > 0.9993 && checkGradient(pos, surfaceHeightFacet)) {
-                Lake temp = new Lake(pos, 10 + 20 * Math.abs(Math.round(noise.noise(pos.x(), pos.z()))));
+            float noiseValue;
+            if (biomeFacet.getWorld(pos.x(), pos.z()).getId().equals(CoreBiome.DESERT.getId())) {
+                noiseValue = noise.noise(pos.x() * 0.1f, pos.y(), pos.z() * 0.1f);
+            } else {
+                noiseValue = noise.noise(pos.x() * 0.5f, pos.y(), pos.z() * 0.5f);
+            }
```

Comment:
This piece of the code looks alright, but I'd like to make a couple of suggestions to improve readability and make life simpler for further extensions.

- rename `noiseValue` >>> `probablity` to make it clearer what this value is used for
- extract the calculation (with the `if-else`) to a private method which gets the position and the biome facet to compute the probability based on that:
  ```java
  float probability = computeProbabilityFor(pos, biomeFacet);
  ```
- I'd already take the lookup of the biome id out into a local variable to so that the comparison becomes a bit shorter. It also allows to use the derived biome for other `else if` checks easily
  ```java
  String biomeId = biomeFacet.getWorld(pos.x(), pos.z()).getId();
  float probability;
  if (biomeId.equals(CoreBiome.DESERT.getId())) {
	probability= noise.noise(pos.x() * 0.1f, pos.y(), pos.z() * 0.1f);
  } else if (biomeId.equals(...)) {
	// not now, but for future extensions
  } else {
	probability= noise.noise(pos.x() * 0.5f, pos.y(), pos.z() * 0.5f);
  }
  return probability;
  ```

Reply:
sounds great, i'll do this